cmd/restic/cmd_prune.go

```diff
@@ -273,11 +273,23 @@ func pruneRepository(gopts GlobalOptions, repo restic.Repository) error {
 
 	removePacks.Merge(obsoletePacks)
 
-	if err = rebuildIndex(ctx, repo, removePacks); err != nil {
-		return err
-	}
-
 	if len(removePacks) != 0 {
+		Verbosef("rebuilding index files...\n")
```

Comment:
I've noticed that the index rebuild is missing a progress bar, please add one (Maybe based on the number of packs which are already added?).

Reply:
It would also be good to extract these 25 added lines into a separate function. `pruneRepository` is already much much much too long.